rs/system_api/tests/sandbox_safe_system_state.rs

```diff
@@ -349,6 +349,48 @@ fn mint_cycles_fails_caller_not_on_nns() {
     );
 }
 
+#[test]
+fn mint_cycles_very_large_value() {
+    let cycles_account_manager = CyclesAccountManagerBuilder::new()
+        .with_subnet_type(SubnetType::System)
+        .build();
+    let mut system_state = SystemStateBuilder::new()
+        .canister_id(CYCLES_MINTING_CANISTER_ID)
+        .build();
+
+    system_state.add_cycles(
+        Cycles::from(1_000_000_000_000_000_u128),
+        CyclesUseCase::NonConsumed,
+    );
+
+    let api_type = ApiTypeBuilder::build_update_api();
+    let mut api = get_system_api(api_type, &system_state, cycles_account_manager);
+    let mut balance_before = [0u8; 16];
+    api.ic0_canister_cycle_balance128(0, &mut balance_before)
+        .unwrap();
+    let balance_before = u128::from_le_bytes(balance_before);
+
+    let amount_high = u64::MAX;
+    let amount_low = 50;
```

Comment:
What if this is u64::MAX, too? This should still work if the initial balance is zero, but not otherwise. So I'd suggest a pair of tests:
- initial balance of zero, minting u128::MAX => success
- initial balance of one, minting u128::MAX => failure

Reply:
The implementation of `CyclesAccountManager::mint_cycles` suggests that both cases succeed (with saturating cycles at u128::MAX in the second case) - still worthwhile having a test (in particular, w.r.t. UTOPIA) and also updating the doc comment on the return value (the system API doesn't return how many cycles were added, but just reflects back the input).

Reply:
Even more preferrably, could we fix the return value of `mint_cycles` and `mint_cycles128` to follow the doc comment: i.e., return the actual amount of cycles that were minted?